cudapoa/CMakeLists.txt

```diff
@@ -26,8 +26,12 @@ cuda_add_library(cudapoa ${cga_library_type}
     )
 
 target_link_libraries(cudapoa utils logging)
+target_compile_options(cudapoa PRIVATE -Werror -Wall -Wextra)
 
-target_compile_options(cudapoa PRIVATE -Werror -Wall -Wextra -march=native)
+option(optimize_for_native "Build with march=native" ON)
```

Comment:
I think it would be better to move this option up to the top level CMakeLists.txt, and call it `cga_optimize_for_native_cpu` (following convention here https://github.com/clara-genomics/ClaraGenomicsAnalysis/blob/dev-v0.5.0/CMakeLists.txt#L17 ). Can you also see if other parts of the system use `march=native` and if so guard them with this option as well?